autotest/ogr/ogr_gpkg.py

```diff
@@ -8451,3 +8451,70 @@ def test_ogr_gpkg_ogr_layer_Extent():
 
     finally:
         gdal.Unlink(tmpfilename)
+
+
+###############################################################################
+# Test field alternative names
+
+
+def test_ogr_gpkg_field_alternative_names():
+
+    dbname = "/vsimem/ogr_gpkg_alternative_names.gpkg"
+    ds = gdaltest.gpkg_dr.CreateDataSource(dbname)
+    lyr = ds.CreateLayer("test", geom_type=ogr.wkbPolygon)
+    lyr.CreateField(ogr.FieldDefn("foo", ogr.OFTString))
+    lyr.CreateField(ogr.FieldDefn("baz", ogr.OFTString))
+
+    # with no gpkg_data_columns table
+    lyr = ds.GetLayer("test")
+    assert lyr.GetLayerDefn().GetFieldCount() == 2
+    assert lyr.GetLayerDefn().GetFieldDefn(0).GetName() == "foo"
+    assert lyr.GetLayerDefn().GetFieldDefn(0).GetAlternativeName() == ""
+    assert lyr.GetLayerDefn().GetFieldDefn(1).GetName() == "baz"
+    assert lyr.GetLayerDefn().GetFieldDefn(1).GetAlternativeName() == ""
+
+    ds.ExecuteSQL(
+        """CREATE TABLE gpkg_data_columns (
+  table_name TEXT NOT NULL,
+  column_name TEXT NOT NULL,
+  name TEXT UNIQUE,
+  title TEXT,
+  description TEXT,
+  mime_type TEXT,
+  constraint_name TEXT,
+  CONSTRAINT pk_gdc PRIMARY KEY (table_name, column_name),
+  CONSTRAINT fk_gdc_tn FOREIGN KEY (table_name) REFERENCES gpkg_contents(table_name)
+)"""
+    )
+    # name and title same as column name, won't be used as alternative name
+    ds.ExecuteSQL(
+        "INSERT INTO gpkg_data_columns('table_name', 'column_name', 'name') VALUES ('test', 'foo', 'foo')"
+    )
+    ds.ExecuteSQL(
+        "INSERT INTO gpkg_data_columns('table_name', 'column_name', 'title') VALUES('test', 'baz', 'baz')"
+    )
```

Comment:
That, and other places testing the baz column to be removed if we handle only name as the possibility for the alias. Cf my general remark in https://github.com/OSGeo/gdal/pull/7579#issuecomment-1502999339

```suggestion
    ds.ExecuteSQL(
        "INSERT INTO gpkg_data_columns('table_name', 'column_name', 'title') VALUES('test', 'baz', 'baz')"
    )
```